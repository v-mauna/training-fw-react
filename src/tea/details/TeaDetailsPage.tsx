import React, { useEffect, useState } from 'react';
import { useParams } from 'react-router';
import {
  IonBackButton,
  IonButtons,
  IonContent,
  IonHeader,
  IonImg,
  IonPage,
  IonTitle,
  IonToolbar,
} from '@ionic/react';
import { useTea } from '../useTea';
import { Tea } from '../../shared/models';

const TeaDetailsPage: React.FC = () => {
  const { id } = useParams<{ id: string }>();
  const { getTeaById } = useTea();
  const [tea, setTea] = useState<Tea | undefined>(undefined);

  useEffect(() => {
    const init = async () => {
      const tea = await getTeaById(parseInt(id, 10));
      setTea(tea);
    };
    init();
  }, [id, getTeaById]);

  return (
    <IonPage>
      <IonHeader>
        <IonToolbar>
          <IonButtons slot="start">
            <IonBackButton defaultHref="/tea" />
          </IonButtons>
          <IonTitle>Details</IonTitle>
        </IonToolbar>
      </IonHeader>
      <IonContent>
        <IonHeader collapse="condense">
          <IonToolbar>
            <IonTitle size="large">Details</IonTitle>
          </IonToolbar>
        </IonHeader>
        <div className="ion-padding">
          <div className="ion-justify-content-center">
            <IonImg src={tea?.image} />
          </div>
          <h1>{tea?.name}</h1>
          <p>{tea?.description}</p>
        </div>
      </IonContent>
    </IonPage>
  );
};
export default TeaDetailsPage;
