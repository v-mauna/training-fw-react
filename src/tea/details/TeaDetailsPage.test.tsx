import React from 'react';
import { render, wait } from '@testing-library/react';
import TeaDetailsPage from './TeaDetailsPage';
jest.mock('react-router', () => ({
  useParams: () => ({
    id: 1,
  }),
}));
jest.mock('../useTea', () => ({
  useTea: () => ({
    getTeas: jest.fn(),
    getTeaById: jest.fn(() => Promise.resolve(mockTea)),
  }),
}));

const mockTea = {
  id: 1,
  name: 'Green',
  image: require('../../assets/images/green.jpg'),
  description:
    'Green teas have the oxidation process stopped very early on, leaving them with a very subtle flavor and ' +
    'complex undertones. These teas should be steeped at lower temperatures for shorter periods of time.',
};

describe('<TeaDetailsPage />', () => {
  it('displays the header', async () => {
    const { container } = render(<TeaDetailsPage />);
    await wait(() => expect(container).toHaveTextContent(/Details/));
  });

  it('renders consistently', async () => {
    const { asFragment } = render(<TeaDetailsPage />);
    await wait(() => expect(asFragment()).toMatchSnapshot());
  });

  it('renders the tea name', async () => {
    const { container } = render(<TeaDetailsPage />);
    await wait(() => expect(container).toHaveTextContent(mockTea.name));
  });

  it('renders the tea description', async () => {
    const { container } = render(<TeaDetailsPage />);
    await wait(() => expect(container).toHaveTextContent(mockTea.description));
  });
});
