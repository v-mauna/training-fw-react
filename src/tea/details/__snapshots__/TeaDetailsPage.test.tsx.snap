// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeaDetailsPage /> renders consistently 1`] = `
<DocumentFragment>
  <div
    class="ion-page"
  >
    <ion-header>
      <ion-toolbar>
        <ion-buttons
          slot="start"
        >
          <ion-back-button
            default-href="/tea"
          />
        </ion-buttons>
        <ion-title>
          Details
        </ion-title>
      </ion-toolbar>
    </ion-header>
    <ion-content>
      <ion-header
        collapse="condense"
      >
        <ion-toolbar>
          <ion-title
            size="large"
          >
            Details
          </ion-title>
        </ion-toolbar>
      </ion-header>
      <div
        class="ion-padding"
      >
        <div
          class="ion-justify-content-center"
        >
          <ion-img
            src="green.jpg"
          />
        </div>
        <h1>
          Green
        </h1>
        <p>
          Green teas have the oxidation process stopped very early on, leaving them with a very subtle flavor and complex undertones. These teas should be steeped at lower temperatures for shorter periods of time.
        </p>
      </div>
    </ion-content>
  </div>
</DocumentFragment>
`;
